### ISSUE TITLE
Public path and output path do not work together?

### ISSUE THREAD

**Issue by HunterGraubard** [webpack-contrib/file-loader#269]
I'm using the latest version of `file-loader`. All I want to do is pull some images from a `node_modules` package and emit it to my output directory. But I've been struggling with this for hours.

I have the following directories involved in this:

* `/assets/css/`
* `/assets/misc/`

When I find `example.png` in `/assets/css/main.css`, I want `file-loader` to copy that file to `/assets/misc/hash.png` and generate the URL to the CSS file as `../misc/example.png`.

```
loader: 'file-loader',
options: {
	outputPath: 'assets/misc/',
	publicPath: '../misc/'
}
```

When I try the above configuration, the generated URL is correct, but no files are actually emitted anywhere - `outputPath` is ignored! If I remove the `publicPath` line, then the files are correctly emitted to `/assets/misc/`, but the generated URL is incorrect (`assets/misc/example.png` instead of `../misc/example.png`.

I've tried alternative solutions as well:

```
loader: 'file-loader',
options: {
	name: baseWebsiteUrl + '[path][hash].[ext]',
	outputPath: 'assets/misc/',
}
```

When I try the above configuration (with `baseWebsiteUrl` being a string containing my website's base URL), the files are again correctly emitted to `/assets/misc/`, but the generate URL is still wrong - this time it's `assets/misc/MYWEBSITEURL/node_modules/NODEPACKAGE/example.png`.

Why can't I simply use `publicPath` and `outputPath` together? Surely this is a bug?

**Comment by HunterGraubard** [comment]
Seems like it's working now. No idea why.

**Comment by Annilla** [comment]
@HunterGraubard 
So which config can work? 
Can you post the final config setting?
I am confused with similar problem like this, too.

**Comment by HunterGraubard** [comment]
Sorry, I don't really understand it myself. I've found this plugin very confusing to use.

```
test: /some-filename-from-node-package\.png$/,
use: [{
	loader: 'file-loader',
	options: {
		publicPath: '/assets/misc/',
		outputPath: 'assets/misc/',
	}
}]
```

I ended up using this configuration. It writes the files to the assets/misc directory, and writes the URL into the stylesheet with a leading / to make it root-relative. Using a `baseWebsiteUrl` variable at the start of `publicPath` is working too now (e.g. `publicPath: baseWebsiteUrl + 'assets/misc/'` writes an absolute URL into the stylesheet).

My main problem before was that when I used `publicPath`, no files would be emitted, making it pointless. But today it's suddenly working, so the problem is gone.
